azavea/iow-boundary-tool
# Setup staging

## Overview

Create the staging environment. The example endpoint and health check are both accessible at the staging url

Connects #8 

### Demo

```
$ http https://staging.iow.azavea.com/api/example/
HTTP/1.1 200 OK
Connection: keep-alive
Content-Length: 20
Content-Security-Policy: frame-ancestors 'none'
Content-Type: text/plain; charset=UTF-8
Date: Wed, 20 Jul 2022 14:24:59 GMT
Referrer-Policy: same-origin
Server: gunicorn
Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
Via: 1.1 6b40574acc577d1185c505c40886acc6.cloudfront.net (CloudFront)
X-Amz-Cf-Id: eOVkYXbPU7WEfvMVh33SOQGxUM3x_8i4r2S70JxTCDhnyQXgxxbC7g==
X-Amz-Cf-Pop: EWR52-C2
X-Cache: Miss from cloudfront
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block

Example API response
```

```
$ http https://staging.iow.azavea.com/health-check/
HTTP/1.1 200 OK
Connection: keep-alive
Content-Length: 42
Content-Security-Policy: frame-ancestors 'none'
Content-Type: application/json
Date: Wed, 20 Jul 2022 14:26:43 GMT
Referrer-Policy: same-origin
Server: gunicorn
Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
Via: 1.1 7f0fedc5c1b6155cbeb8fd32bdcdb07a.cloudfront.net (CloudFront)
X-Amz-Cf-Id: NVx6agXOrKTwudKHZ6fDpS7FT3jqYfUJxqMqx7xAPMDOGOcX3WX13w==
X-Amz-Cf-Pop: EWR52-C2
X-Cache: Miss from cloudfront
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block

{
    "databases": [
        {
            "default": {
                "ok": true
            }
        }
    ]
}
```

## Testing Instructions
* confirm that you get a response from https://staging.iow.azavea.com/api/example/ 
* confirm that the health check loads https://staging.iow.azavea.com/health-check/
